sample_files/main.py

```diff
@@ -13,43 +16,68 @@
 
 # Initialize logging
 logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
 
 # Load environment variables
-from dotenv import load_dotenv
-import os
-
 load_dotenv()
 
 # Fetch API key
 api_key = os.getenv('OPENAI_API_KEY')
 
+if not api_key:
+    logger.error("OPENAI_API_KEY environment variable not found")
+    raise EnvironmentError("OPENAI_API_KEY environment variable not found")
+
 # Initialize OpenAI Chat model
 llm = ChatOpenAI(temperature=0.8, api_key=api_key)
 
-# Initialize Chrome WebDriver using WebDriver Manager
 def get_webdriver():
-    return webdriver.Chrome(ChromeDriverManager().install())
+    """
+    Initialize Chrome WebDriver using WebDriver Manager.
+    """
+    try:
+        return webdriver.Chrome(ChromeDriverManager().install())
+    except WebDriverException as e:
+        logger.error(f"Failed to initialize WebDriver: {e}")
+        raise
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

LGTM!


<!-- This is an auto-generated comment by OSS CodeRabbit -->